src/Symfony/Component/Translation/DataCollectorTranslator.php

```diff
@@ -25,12 +25,9 @@ class DataCollectorTranslator implements TranslatorInterface, TranslatorBagInter
     public const MESSAGE_MISSING = 1;
     public const MESSAGE_EQUALS_FALLBACK = 2;
 
-    /**
-     * @var TranslatorInterface|TranslatorBagInterface
-     */
-    private $translator;
+    private TranslatorInterface | TranslatorBagInterface $translator;
```

Comment:
the type should actually be `TranslatorInterface & TranslatorBagInterface & LocaleAwareInterface`, not a union type (see the constructor). So in the meantime, I would use `TranslatorInterface` as the property type (and maybe use phpdoc for the actual intersection type).

Also, the phpdoc should be fixed in older branches to use an intersection type.

Reply:
fixed in bd15e17227f4d6847df85b093d55e28f3ddccbf3